BEP7.md

```diff
@@ -0,0 +1,320 @@
+# BEP-7: Non-fungible Tokens on Binance Chain
+
+- [BEP-7: Non-fungible Tokens on Binance Chain](#bep-7-non-fungible-tokens-on-binance-chain)
+  - [1. Summary](#1-summary)
+  - [2. Abstract](#2-abstract)
+  - [3. Status](#3-status)
+    - [Couple things to keep in mind](couple-things-to-keep-in-mind)
+  - [4. Motivation](#4-motivation)
+  - [5. Collection Specification](#5-collection-specification)
+    - [5.1 Collection Properties](#51-collection-properties)
+    - [5.2 Collection Management Operation](#52-collection-management-operation)
+      - [5.2.1 Issue collection](#521-issue-collection)
+      - [5.2.2 Increase total supply](#522-increase-total-supply)
+  - [6. Token Specification](#6-token-specification)
+    - [6.1 Token Properties](#61-token-properties)
+    - [6.2 Token Management Operation](#62-token-management-operation)
+      - [6.2.1 Issue token](#621-issue-token)
+      - [6.2.2 Transfer Tokens](#622-transfer-tokens)
+      - [6.2.3 Freeze Tokens](#623-freeze-tokens)
+      - [6.2.4 Unfreeze Tokens](#624-unfreeze-tokens)
+  - [7. Token Use Case Examples](#7-token-use-case-examples)
+  - [8. License](#8-license)
+  - [9. Reference](#9-reference)
+
+## 1. Summary
+
+This BEP describes a proposal for non-fungible tokens(NFTs) management on the Binance Chain.
+
+## 2. Abstract
+
+BEP-7 Proposal describes a common set of rules for non-fungible token within the Binance Chain ecosystem. It introduces the following details of a non-fungible token on Binance Chain:
+
+- What information makes a non-fungible token on Binance Chain
+- What actions can be performed on a non-fungible token on Binance Chain
+
+### Couple things to keep in mind
+
+- This BEP won't contain "how to implement" in terms of implmentation on the Binance Chain codebase, e.g. which module NFT should be in (this kind of discussion is happening on [cosmos-sdk/issues/4046](https://github.com/cosmos/cosmos-sdk/issues/4046)), because that should be happening in the PR(against to Binance Chain, when it's fully open source, so that dev can send out PR.
+
+- Apparently Binance chain doens't have solidity-such smart contract implemented, so don't expect to see many smart contract implmenations like what you can see in [EIP-721](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-721.md) in this BEP.
+
+## 3. Status
+
+This BEP is under drafting.
+
+## 4. Motivation
+
+Design and issue non-fungible asset on the Binance Chain, as one of the basic economic foundations of the blockchain.
+
+## 5. Collection Specification
```

Comment:
Chain ID List mutate state NFT HUB enforcement that enables Mutation -ZManian

Reply:
So there are two good ways of doing mutable NFAs in an IBC system.

The first approach requires all mutations to send IBC messages back to the issuing chain and it returns the result of the mutation.

The other solution is to have the issuer provide a set of chains whose mutations it will treat as cannonical and thus given a hub that enforces this white list would allow everyone else to treat those mutations as cannonical as well without the round trips to the issuer zone

Reply:
Are the mutations we're referring to with regard to metadata or to ownership?

If wrt metadata then the whitelist scenario might result in multiple sources of truth for that metadata--informationn stored on the origin chain and information stored on the secondary chain. This would mean the origin chain would need to monitor changes to metadata instead of the secondary chain.

Since monitoring changes to metadata must take place regardless I'd suggest that it is the secondary chain who must monitor changes on the origin chain. A whitelist could be managed to determine if the secondary chain has permissions to modify aspects of the metadata on the origin chain, but for simplicity and sanity it seems that the origin chain should be the source of truth of the metadata. The secondary chain should be able to implement their own system for tracking that metadata, either keeping a local version in sync or querying the origin chain when necessary.

Reply:
NFTs
IBC format for NFTs. Billy working on NFT-IBC spec. NFT will follow after the token transfer. 
Current design: NFT will not carry metadata with them. Only have identity and chain ID.
Right now the way it works in Ethereum, there is one field that is very short that can be resolved to get metadata. Billy is wondering whether to continue with that approach, or have the metadata on chain. The ability to update the metadata should live with the zone, rather than transfer over IBC. When you are transferring NFTs over IBC, you would only transfer the denomination - you know where the origin chain is, and you can go a query that chain for the metadata.
Zaki: Being able to have a white list of multiple zones that are allowed to mutate the metadata is going to be valuable. Instead of saying, just transfer the ID, you can transfer the metadata along with a white list of zones that have the ability to change the metadata. Hub can enforce that only metadata updates from white listed chains are persistent.
A use case for the above: Breeding  crypto kitties example. If the metadata can only be changed by issuing chain, then need to return the kitty every time to make a change to it. A white list would allow other zones to also breed crypto kitties.
NFT spaces being driven by gaming. Financial derivative NFTs should have a different spec. The same spec may not suit gaming and financial NFTs.
Question 1 from Meher: How is the whitelist made In the case of NFT staking derivatives where the issuer is the protocol?
Question 2 from Meher: If multiple genes can mutate the metadata of an NFT, and many of these mutations happen for the same NFT on different chains, won’t the trustworthiness of the final state of the NFT be dependent on these security of all of these chains? and this dependence would imply that it security is very low?
Zaki: Small number of zones that can make changes to the NFT metadata.  this is like a contract between Chains,  where the issuing chain issues the right to mutate metadata to a few number of chains that are near it in the IBC graph. Issuing zone can effectively debase a NFT tokens, thus always keeping the power to recover from bad mutations.
The issuing chain is pretty special and lots of things have to go back to that. 
There are also around ideas such that: Original issuance of NFTs can be someone other zone, but the hub ensures some kind of security properties of the NFTs.
These designs leading to a world where there is an authoritative source, and other somewhat less authoritative sources. As a user you can figure out the relationship, and refer back to the authoritative Zone when needed. There may be some advantages of working with the less authoritative version, as it might have better transaction properties or better connectivity.
Decision: Sum all this up on the NFT ICS issue.

Reply:
Yes I agree that there should be the option for external chains to update the metadata on the origin chain. A white list would be a convenient method but I think there may be more complicated systems needed by some chains and it should be up to the origin chain to implement as simple or as complicated of a system as they need. A white list would be a great solution for simple cases and might be enough for Binance Chain.

Reply:
NFT Module intergrated with the Cosmos SDK v.0 v.1 
-Metadata Interface 
-Metadata Struct includes the interface for offchain key-value pairs
-Introspection -description JSON that confirms the Attribute(s)

IBC Standard
-ID includes tokenID
-Denom includes demonination

IBC NFT Transfer x origin chain
-ID tokenID
-originChain B.originChain 
-Metadata on wheter or not it sold on Binance Dex
-approval vs transfer in regards to cross-chain transfer of ownership delegation of utility in regards to its readability by wallets regarding its fungibility status during that time period whether it is meant to be a BEP2/ERC-20 token or semi-fungible ERC-1155 in such case that it is a collection of ERC-721 unique one of a kind NFT or an (index)NFTarray[]



ERC-1155 interoperability with Enjin Coin Unity SDK

dGoods Mythical Goods EOS 
-Blockchain Game developers 
-Schema.org Web3 W3c off-chain consortium (metadata)
-github